src/main/java/rx/Scheduler.java

```diff
@@ -119,11 +120,17 @@ public void call() {
                     if (!mas.isUnsubscribed()) {
                         action.call();
                         long nextTick = startInNanos + (++count * periodInNanos);
-                        mas.set(schedule(this, nextTick - TimeUnit.MILLISECONDS.toNanos(now()), TimeUnit.NANOSECONDS));
+                        SerialSubscription s = new SerialSubscription();
```

Comment:
Do you mean if `period` is large, the new Subscription created in the next action won't be able to replace the one here, and if `period` is small, the leak won't be a problem?

Just to make sure we are in the same page.